synapse/api/ratelimiting.py

```diff
@@ -43,6 +44,23 @@ def __init__(self, clock: Clock, rate_hz: float, burst_count: int):
         #   * The rate_hz of this particular entry. This can vary per request
         self.actions = OrderedDict()  # type: OrderedDict[Any, Tuple[float, int, float]]
 
+    def can_requester_do_action(
```

Comment:
I wonder if this logic should just be in `can_do_action`?

Reply:
There appeared to be some usages of can_do_action which didn't use a requester object, so I split it out to be safe.